2016-04/april-7.md

```diff
@@ -0,0 +1,129 @@
+
+#### Reasons to Use Properties over Attributes
+
+* Attributes seem to only match our wanted behavior for “metadata”, i.e. things that the DOM doesn’t actually really care about. The exception is SVG.
+* Attributes for initial render is a waste of resources (`toString` / `parse`) but semantically is OK. They also make sense for server rendering since the point of attributes is to provide an initial value. However this is not a good argument for why they should be used for values that change over time.
+* The addition of new elements and features leads to more controlled values and richer data types. Attributes are the legacy, not properties.
+* If there are any semantic differences that depend on attributes, we should flag those to standard committees. Then we can polyfill those using attributes, and then eventually remove the fix.
+* Maybe we should find someone to actually design this rich API mapping for React DOM props and maintain it. We aren’t busy with this now but this doesn’t mean the problem isn’t worthwhile. Hopefully one day it might be standardized in some form or just considered the canonical declarative bindings to the DOM.
+
+#### Conclusion
+
+* No conlusion yet. We plan to come back to this next week.
+* React 15 uses attributes.
+* The regressions in RCs appear fixed.
+
+#### Relevant Issues and Pull Requests
+
+* [#1431](https://github.com/facebook/react/issues/1431)
+* [#1510](https://github.com/facebook/react/pull/1510)
+* [#4618](https://github.com/facebook/react/issues/4618)
+* [#5666](https://github.com/facebook/react/pull/5666)
+* [#5680](https://github.com/facebook/react/pull/5680)
+* [#5907](https://github.com/facebook/react/pull/5907)
+* [#5966](https://github.com/facebook/react/issues/5966)
+* [#6119](https://github.com/facebook/react/issues/6119)
+* [#6219](https://github.com/facebook/react/issues/6219)
+* [#6228](https://github.com/facebook/react/pull/6228)
+* [#6406](https://github.com/facebook/react/pull/6406)
+
+### Recent Pull Requests
+
+#### `ReactDOM.render()` return value being <s>deprecated</s> legacy ([#6400](https://github.com/facebook/react/pull/6400))
+
+* It’s not deprecated yet and we don’t have a solid plan. But new code should avoid it.
+* We still rely on it heavily in our tests (basically all of them!)
+  * We might need a separate reconciler for tests anyway.
+* We want to encourage people to move away from it because, if we implement incremental reconciler, we might not be able to always resolve nodes synchronously.
+* The only thing we know for sure is that this is a potential stumbling block going forward.
+
+#### [Sebastian](https://twitter.com/sebmarkbage) is moving some files from React Native to React ([#6338](https://github.com/facebook/react/pull/6338))
+
+* This adds a new package called `react-native-renderer`.
+* It doesn’t block 15 and won’t need to wait until 16.
+* We might also split out the `react-dom-renderer` from `react-dom` in the future.
```

Comment:
This was addressed here: https://github.com/facebook/react/issues/6169#issuecomment-191879640. Ideally factories should be moved to a completely different package because they aren't used by most people and don't actually depend on specific renderer.